custom_components/elasticsearch/datastreams/index_template.json

```diff
@@ -0,0 +1,338 @@
+{
+    "index_patterns": [
+        "metrics-homeassistant.*.events-default*"
+    ],
+    "template": {
+        "mappings": {
+            "dynamic": "strict",
+            "dynamic_templates": [
+                {
+                    "hass_attributes": {
+                        "path_match": "hass.attributes.*",
+                        "mapping": {
+                            "type": "text",
+                            "fields": {
+                                "keyword": {
+                                    "ignore_above": 1024,
+                                    "type": "keyword"
+                                },
+                                "float": {
+                                    "ignore_malformed": true,
+                                    "type": "float"
+                                }
+                            }
+                        }
+                    }
+                },
```

Comment:
Agree, same for the hass.entity.attributes too?

Reply:
I believe it still makes sense to keep `hass.entity.attributes`. We "moved" `hass.attributes` into `hass.entity.attributes`, but kept the old one around so that we didn't break anyone who was relying on those existing fields. With the migration to data streams, we can make a clean break and ask folks to use the new location at `hass.entity.attributes`

Reply:
I see three sets that are duplicated:
entity.id and entity_id
hass.domain and hass.entity.domain
hass.attributes and hass.entity.attributes

I could modify the doc creator to have a format version, and have legacy use format 1 where we keep all 3 duplicates and datastreams use format 2 where we could remove all 3 duplicates.

If we want to maintain backwards compatibility we can setup aliases in the datastream index template to alias hass.entity_id and hass.domain but we wouldn't be able to alias hass.attributes which it sounds like you're okay with?

So to summarize I propose in order of preference:
1. Remove all 3 duplicates
2. Alternatively, remove all 3 duplicates but provide backwards compat for hass.entity_id and hass.domain via aliases in the index template.

Thoughts?

Reply:
Implemented this without aliases, can undo based on feedback

Reply:
> Implemented this without aliases, can undo based on feedback

I'm happy with this approach, thanks